rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -455,84 +461,28 @@ def _tag_id_index_mapping_for(
 
         return tag_id_dict
 
-    @staticmethod
-    def _find_example_for_label(
-        label: Text, examples: List[Message], attribute: Text
-    ) -> Optional[Message]:
-        for ex in examples:
-            if ex.get(attribute) == label:
-                return ex
-        return None
-
-    def _check_labels_features_exist(
-        self, labels_example: List[Message], attribute: Text
-    ) -> bool:
-        """Checks if all labels have features set."""
-
-        return all(
-            label_example.features_present(
-                attribute, self.component_config[FEATURIZERS]
-            )
-            for label_example in labels_example
-        )
-
-    def _extract_features(
-        self, message: Message, attribute: Text
-    ) -> Dict[Text, Union[scipy.sparse.spmatrix, np.ndarray]]:
+    def _uses_sequence_features_for_text(self) -> bool:
+        """Whether we need sequence features for the `TEXT` attribute.
 
-        (
-            sparse_sequence_features,
-            sparse_sentence_features,
-        ) = message.get_sparse_features(attribute, self.component_config[FEATURIZERS])
-        dense_sequence_features, dense_sentence_features = message.get_dense_features(
-            attribute, self.component_config[FEATURIZERS]
+        Note that, for prediction of some label attribute, the `DIETClassifier`
+        can make use of sequence features even if the number of
+        transformer layers is 0 because for that it creates a BOW representation
+        by simply summing over sequence+sentence features.
```

Comment:
No I mean[ the summation for the BOW](https://github.com/RasaHQ/rasa/blob/f78a6a17d5146248500630a2397ee099a24e0f5c/rasa/nlu/classifiers/diet_classifier.py#L1451). 
At the moment (not saying this is useful or should happen), they make use of them (i.e. their presence influences the output) because they sum them all up, despite the sequence features not begin passed through any transformer. In contrast to any model that would just use the last token of the concatenated sequence (after that is *not*. being passed through a transformer because the number of transformer layers is 0).